regression-test/suites/nereids_rules_p0/salt_join/salt_join.groovy

```diff
@@ -0,0 +1,282 @@
+    sql """insert into test_skew10 values(1,'abc',9),(null,'abc1',3),(1,'def',2),(2,'def1',7),(2,'def',2),(3,'abc',4),(5,'def',8),(2,'def',2),(5,'def',null),(3,'abc',null)"""
+    qt_simple """
+    select * from test_skew9 tl inner join [shuffle(skew(tl.b(1,2)))] test_skew10 tr on tl.b = tr.b order by 1,2,3,4,5,6;
+    """
+    qt_left_simple """
+    select * from test_skew9 tl left join [shuffle(skew(tl.b(1,2)))] test_skew10 tr on tl.b = tr.b order by 1,2,3,4,5,6;
+    """
+    qt_right_simple """
+    select * from test_skew9 tl right join [shuffle(skew(tl.b(1,2)))] test_skew10 tr on tl.b = tr.b order by 1,2,3,4,5,6;
+    """
+    qt_skew_value_null """
+    select * from test_skew9 tl inner join [shuffle(skew(tl.b(1,2,null)))] test_skew10 tr on tl.b = tr.b order by 1,2,3,4,5,6;
+    """
+    qt_skew_value_only_null """
+    select * from test_skew9 tl inner join [shuffle(skew(tl.b(null)))] test_skew10 tr on tl.b = tr.b order by 1,2,3,4,5,6;
+    """
+    qt_leading """
+    select /*+leading(tl shuffle (skew(tl.b(1,2))) tr) */ * from test_skew9 tl join test_skew10 tr on tl.b=tr.b order by 1,2,3,4,5,6;
+    """
+    qt_has_other_equal_condition """
+    select /*+leading(tl shuffle (skew(tl.b(1,2))) tr) */ * from test_skew9 tl join test_skew10 tr on tl.b=tr.b and tl.a=tr.a order by 1,2,3,4,5,6;
+    """
+    qt_has_other_unequal_condition """
+    select /*+leading(tl shuffle (skew(tl.b(1,2))) tr) */ * from test_skew9 tl join test_skew10 tr on tl.b=tr.b and tl.a<tr.a order by 1,2,3,4,5,6;
+    """
+    qt_test_varchar_skew_value """
+    select /*+leading(tl shuffle (skew(tl.c("abc","def"))) tr) */ * from test_skew9 tl join test_skew10 tr on tl.c=tr.c and tl.a=tr.a order by 1,2,3,4,5,6;
+    """
+    qt_test_multi_join """
+    select * from test_skew9 tl inner join [shuffle(skew(tl.b(1,2,null)))] test_skew10 tr on tl.b = tr.b left join  [shuffle(skew(tl.b(1,2,null)))] test_skew10 tt on tl.b=tt.b
+    order by 1,2,3,4,5,6,7,8,9;
+    """
+    qt_test_multi_join_right_join """
+    select * from test_skew9 tl inner join [shuffle(skew(tl.b(1,2,null)))] test_skew10 tr on tl.b = tr.b right join  [shuffle(skew(tl.b(1,2,null)))] test_skew10 tt on tl.b=tt.b
+    order by 1,2,3,4,5,6,7,8,9;
+    """
+    qt_test_left_join_right_join """
+    select * from test_skew9 tl left join [shuffle(skew(tl.b(1,2,null)))] test_skew10 tr on tl.b = tr.b right join  [shuffle(skew(tl.b(1,2,null)))] test_skew10 tt on tl.b=tt.b
+    order by 1,2,3,4,5,6,7,8,9;
+    """
+    qt_leading_multi_join """
+    select /*+leading(tl shuffle (skew(tl.c("abc","def"))) tr shuffle(skew(tl.c("abc","def"))) tt) */ * from 
+    test_skew9 tl join test_skew10 tr on tl.c=tr.c and tl.a=tr.a inner join test_skew10 tt on tl.c = tt.c
+    order by 1,2,3,4,5,6,7,8,9;
+    """
+    qt_leading_multi_join_bracket """
+    select /*+leading(tl shuffle (skew(tl.c("abc","def"))) {tr shuffle(skew(tr.c("abc","def"))) tt}) */ * from 
```

Comment:
in future, we will use () to specify join order precedence instead of {}, so (skew...) may conflict with that. Can we use [skew...] instead of (skew...)?